apps/server/src/modules/tool/uc/external-tool.uc.ts

```diff
@@ -71,4 +71,18 @@ export class ExternalToolUc {
 			throw new UnprocessableEntityException(`The Client Id of the tool: ${externalToolDO.name} is already used`);
 		}
 	}
+
+	async getExternalTool(userId: string, toolId: string): Promise<ExternalToolDO> {
+		const user: User = await this.authorizationService.getUserWithPermissions(userId);
+		this.authorizationService.checkAllPermissions(user, [Permission.TOOL_ADMIN]);
+
+		const tool: ExternalToolDO = await this.externalToolService.findExternalToolById(toolId);
+
+		if (tool.config instanceof Oauth2ToolConfigDO) {
+			const oauthClient: ProviderOauthClient = await this.oauthProviderService.getOAuth2Client(tool.config.clientId);
+			tool.config = this.externalToolMapper.applyProviderOauthClientToDO(tool.config, oauthClient);
```

Comment:
Altering the state of a domain object should not be the responsibility of a (request-)mapper. This should be done in the domain object itself or (if more general) in a domain service.